### PR TITLE
Updated ARC optimized pooling kernel to use the new TfLiteEvalTensor API.

### LINKED ISSUE #43910: Optimized arc kernels need to use the new TfLiteEvalTensor API
The TFLM team recently ported all kernels except a handful of externally maintained optimized kernels. The new API enables very low memory overhead for TFLM. The primary change is the `TfLiteTensor` C struct is only available during `TfLiteRegistration::Prepare` for a kernel. Those structs are served under temporary memory and all data is only available during the lifetime of that method. All TFLM kernels should request the `TfLiteEvalTensor` C struct during `TfLiteRegistration::Eval` calls. 

A sample change looks like this:
https://github.com/tensorflow/tensorflow/commit/e392050297b62772fb9e6aaf10cf1214cb5261e7

This issue tracks updating the Arc kernels to this new API.

@dzakhar 
@JaccovG

PTAL - happy to review any PRs as they come in.

### PR DESCRIPTION
**Pull request by gerbauz**
This pull request is part of porting ARC optimized kernels to the new TfLiteEvalTensor API.
This one is related to pooling kernel.

PR number: [4/5]

Other related PRs:
1. https://github.com/tensorflow/tensorflow/pull/44617
2. https://github.com/tensorflow/tensorflow/pull/44618
3. https://github.com/tensorflow/tensorflow/pull/44619
4. https://github.com/tensorflow/tensorflow/pull/44620
5. https://github.com/tensorflow/tensorflow/pull/44621

Fixes #43910

### CODE REVIEW

File: tensorflow/lite/micro/kernels/arc_mli/pooling.cc [hunk 1]
```diff
@@ -62,13 +75,75 @@ TfLiteStatus CalculateOpData(const TfLiteContext* context,
...
+}
+
+void* Init(TfLiteContext* context, const char* buffer, size_t length) {
+  TFLITE_DCHECK(context->AllocatePersistentBuffer != nullptr);
+  return context->AllocatePersistentBuffer(context, sizeof(OpData));
+}
+
+TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
+  TFLITE_DCHECK(node->builtin_data != nullptr);
+  auto* params = reinterpret_cast<TfLitePoolParams*>(node->builtin_data);
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/arc_mli/pooling.cc:88]: static_cast?

File: tensorflow/lite/micro/kernels/arc_mli/pooling.cc [hunk 2]
```diff
@@ -62,13 +75,75 @@ TfLiteStatus CalculateOpData(const TfLiteContext* context,
...
+                                      &data->activation_max);
+  }
+
+  if (data->is_mli_applicable) {
+    data->mli_in = reinterpret_cast<mli_tensor*>(
+        context->AllocatePersistentBuffer(context, sizeof(mli_tensor)));
+    data->mli_out = reinterpret_cast<mli_tensor*>(
+        context->AllocatePersistentBuffer(context, sizeof(mli_tensor)));
+    data->cfg = reinterpret_cast<mli_pool_cfg*>(
+        context->AllocatePersistentBuffer(context, sizeof(mli_pool_cfg)));
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/arc_mli/pooling.cc:117]: static_cast

**Review by advaitjain — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/kernels/arc_mli/pooling.cc (hunk 1 above)
Comment (gerbauz) [reply tensorflow/lite/micro/kernels/arc_mli/pooling.cc:88]: https://github.com/tensorflow/tensorflow/blob/7978215e252e786f152504ab8999879aa6850cc6/tensorflow/lite/micro/kernels/pooling.cc#L217-L219

I just made it like it is in the reference kernel, what do you think?

File: tensorflow/lite/micro/kernels/arc_mli/pooling.cc (hunk 2 above)
Comment (gerbauz) [reply tensorflow/lite/micro/kernels/arc_mli/pooling.cc:117]: Fixed and committed.

**Review by advaitjain — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Pull request merged**